20_1 DataGrid onRowUpdating: docissues (#1891)
DevExpress/devextreme-documentation: api-reference/10 UI Components/GridBase/1 Configuration/onRowUpdating.md

The row's old data.

---

> [note]

Reviewer: ```suggestion
[note]

Please review the following notes:
```

> [note]

Reviewer: Move this note to the end.

> - Use this function to perform checks before data is updated. If you need to implement a custom update logic, implement [CustomStore](/Documentation/ApiReference/Data_Layer/CustomStore/)'s [update](Documentation/ApiReference/Data_Layer/CustomStore/Configuration/#update) function.

Reviewer: ```suggestion
- Do not use this function to update data. If you need to implement a custom update logic, implement [CustomStore](/Documentation/ApiReference/Data_Layer/CustomStore/)'s [update](Documentation/ApiReference/Data_Layer/CustomStore/Configuration/#update) function.
```

- In batch [editing mode](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/editing/mode.md '{basewidgetpath}/Configuration/editing/#mode'), this function is executed for each row individually if several rows should be updated.

[note]

> The following code shows how to use the function parameter's **cancel** field to prevent or continue row updating. In this code, a Promise is assigned to this field. Row updating continues if a user confirms updating and row data validation on the server succeeds (the Promise is resolved); otherwise, row updating is prevented (the Promise is rejected).

Reviewer: ```suggestion
This function allows you to intercept row update and perform additional actions. The following code shows how to use the function parameter's **cancel** field to prevent or continue row update. In this code, a Promise is assigned to this field. Row update continues if a user confirms it and row data validation on the server succeeds (the Promise is resolved); otherwise, row update is prevented (the Promise is rejected).
```
